#!/usr/bin/env python3
PACKAGE = "predictive_spliner"
import rospkg
import yaml, os
from dynamic_reconfigure.parameter_generator_catkin import *

ros_path = rospkg.RosPack().get_path('predictive_spliner')


gen = ParameterGenerator()
gen.add("n_time_steps", int_t, 0, "Number of time steps for prediction", 400, 10, 500)
gen.add("dt", double_t, 0, "Time step for prediction", 0.02, 0.01, 0.1)
gen.add("save_distance_front", double_t, 0, "Length of car in the front plus margin for saftey", 0.6, 0.1, 1.0)
gen.add("save_distance_back", double_t, 0, "Length of car in the back plus margin for saftey", 0.6, 0.1, 1.0)
gen.add("max_v", double_t, 0, "Maximum velocity of the car", 10.0, 5.0, 15.0)
gen.add("min_v", double_t, 0, "Minimum velocity of the car", 0.0, 0.0, 5.0)
gen.add("max_a", double_t, 0, "Maximum acceleration of the car", 7.0, 4.0, 20.0)
gen.add("min_a", double_t, 0, "Minimum acceleration of the car", 5.0, 1.0, 7.0)
gen.add("max_expire_counter", int_t, 0, "Maximum n of iterations until collision info gets deleted", 10, 0, 20)

gen.add("update_waypoints", bool_t, 0, "Update waypoints", True)
gen.add("speed_offset", double_t, 0, "Add speed offset", 0.0, -1, 1)


exit(gen.generate(PACKAGE, "dynamic_collision_tuner", "dyn_collision_tuner"))